compiler/modules/types/core.bal

```diff
@@ -581,7 +581,7 @@ public function intersect(SemType t1, SemType t2) returns SemType {
     return memoIntersect((), t1, t2);
 }
 
-function memoIntersect(Context? cx, SemType t1, SemType t2) returns SemType {
+public function memoIntersect(Context? cx, SemType t1, SemType t2) returns SemType {
```

Comment:
memoIntersect should not allow nil Context.

Reply:
fixed